sources/osg/Matrix.js

```diff
@@ -651,12 +692,16 @@ define( [
             var tmpVec = [ 0.0, 0.0, 0.0 ];
 
             return function ( matrix, vector, result ) {
-                var d = 1.0 / ( matrix[ 3 ] * vector[ 0 ] + matrix[ 7 ] * vector[ 1 ] + matrix[ 11 ] * vector[ 2 ] + matrix[ 15 ] );
-
                 if ( result === undefined ) {
-                    Notify.warn( 'no matrix destination !' );
-                    result = Matrix.create();
+                    Notify.warn( 'no vec3 destination !' );
+                    result = Vec3.create();
                 }
+                if ( Matrix.isIdentity( matrix ) ) {
```

Comment:
Ouch, debug code. Ought to be a lot of them.

*VERY* useful/interesting when debugging matrix (and transform/stack/visitor/etc...) so that you gain time by only stepping into matrix compute when it does something....

Doesn't slow thing from what I noticed but not benched, Could be that we make an issue to check at it when we have a bench Utilitary thing. Could even be faster as what it does finally is a kind of prefetch. (wish there was a native array compare that would do a memcmp)

Could bench that, with a special case for where a scene is "flattened", therefore all matrice but view is an identity...

Only place where I would let it is in the NodePath matrix building, where getting a nodepath full of identity matrix ought to be slower not matter what you do. 
( https://github.com/cedricpinson/osgjs/blob/491237d01fa4f2dbc9914089adc16aef30fdcb47/sources/osg/Node.js#L257 etc.)

So I remove them all but in nodepath building and make an issue for further investigation ?
(for when we have bench and a FlattenVisitor )